src/main/java/com/scylladb/cdc/debezium/connector/ScyllaSchema.java

```diff
@@ -179,21 +185,51 @@ private Schema computeColumnSchema(ChangeSchema.ColumnDefinition cdef) {
                 return Schema.OPTIONAL_INT8_SCHEMA;
             case DURATION:
                 return Schema.OPTIONAL_STRING_SCHEMA;
-            case LIST:
-            case MAP:
             case SET:
-            case UDT:
-            case TUPLE:
+            case LIST: {
+                Schema innerSchema = computeColumnSchema(type.getTypeArguments().get(0));
+                return SchemaBuilder.array(innerSchema);
+            }
+            case MAP: {
+                Schema keySchema = computeColumnSchema(type.getTypeArguments().get(0));
+                Schema valueSchema = computeColumnSchema(type.getTypeArguments().get(1));
+                return SchemaBuilder.map(keySchema, valueSchema);
+            }
+            case TUPLE: {
+                List<Schema> innerSchemas = type.getTypeArguments().stream()
+                        .map(ScyllaSchema::computeColumnSchema).collect(Collectors.toList());
+                SchemaBuilder tupleSchema = SchemaBuilder.struct();
+                for (int i = 0; i < innerSchemas.size(); i++) {
+                    tupleSchema = tupleSchema.field("tuple_member_" + i, innerSchemas.get(i));
+                }
+                return tupleSchema.optional().build();
+            }
+            case UDT: {
+                SchemaBuilder udtSchema = SchemaBuilder.struct();
```

Comment:
I had to change this line to the code below to make it work with avro:

`                SchemaBuilder udtSchema = SchemaBuilder.struct().name(type.getUdtType().getName());`